tensorflow_addons/layers/normalizations.py

```diff
@@ -321,3 +321,214 @@ def __init__(self, **kwargs):
+        gamma_initializer: Initializer for the gamma weight.
+        beta_regularizer: Optional regularizer for the beta weight.
+        gamma_regularizer: Optional regularizer for the gamma weight.
+        beta_constraint: Optional constraint for the beta weight.
+        gamma_constraint: Optional constraint for the gamma weight.
+        learned_epsilon: (bool) Whether to add another learnable
+        epsilon parameter or not.
+        name: Optional name for the layer
+
+    Input shape
+        Arbitrary. Use the keyword argument `input_shape`
+        (tuple of integers, does not include the samples axis)
+        when using this layer as the first layer in a model. This layer, as of now,
+        works on a 4-D tensor where the tensor should have the shape [N X H X W X C]
+
+        TODO: Add support for more input shapes, especially for FC layers.
+
+    Output shape
+        Same shape as input.
+
+    References
+        - [Filter Response Normalization Layer: Eliminating Batch Dependence
+        in the training of Deep Neural Networks]
+        (https://arxiv.org/abs/1911.09737)
+    """
+
+    def __init__(
+        self,
+        epsilon: float = 1e-6,
+        axis: list = [1, 2],
+        beta_initializer: types.Initializer = "zeros",
+        gamma_initializer: types.Initializer = "ones",
+        beta_regularizer: types.Regularizer = None,
+        gamma_regularizer: types.Regularizer = None,
+        beta_constraint: types.Constraint = None,
+        gamma_constraint: types.Constraint = None,
+        learned_epsilon: bool = False,
+        name: str = None,
+        **kwargs
+    ):
+        super().__init__(name=name, **kwargs)
+        self.epsilon = tf.cast(epsilon, dtype=self.dtype)
+        self.beta_initializer = tf.keras.initializers.get(beta_initializer)
+        self.gamma_initializer = tf.keras.initializers.get(gamma_initializer)
+        self.beta_regularizer = tf.keras.regularizers.get(beta_regularizer)
+        self.gamma_regularizer = tf.keras.regularizers.get(gamma_regularizer)
+        self.beta_constraint = tf.keras.constraints.get(beta_constraint)
+        self.gamma_constraint = tf.keras.constraints.get(gamma_constraint)
+        self.use_eps_learned = learned_epsilon
+        self.supports_masking = True
+
+        if self.use_eps_learned:
+            self.eps_learned_initializer = tf.keras.initializers.Constant(1e-4)
+        else:
+            self.eps_learned_initializer = None
+
+        self._check_axis(axis)
+
+    def build(self, input_shape):
+        if len(tf.TensorShape(input_shape)) != 4:
+            raise ValueError(
+                """Only 4-D tensors (CNNs) are supported
+        as of now."""
+            )
+        self._check_if_input_shape_is_none(input_shape)
+        self._create_input_spec(input_shape)
+        self._add_gamma_weight(input_shape)
+        self._add_beta_weight(input_shape)
+
+        if self.use_eps_learned:
+            self._add_eps_learned_weight()
```

Comment:
This can be moved to the __init__ because we don't depend on the input shape.

Reply:
Cool. Will do that.

Reply:
Done. Thanks